tools/perf_test_cgal.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Emmett Lalish & contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_mesh_processing/corefinement.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+
+#include "manifold.h"
+
+using namespace manifold;
+
+// Epick = Exact predicates Inexact constructions. Seems fair to use to compare
+// to Manifold, which seems to store double coordinates.
+typedef CGAL::Epick Kernel;
+
+// Epeck = Exact predicates Exact constructions. What OpenSCAD uses to guarantee
+// geometry ends up where it should even after many operations. typedef
+// CGAL::Epeck Kernel;
```

Comment:
I should probably read up on CGAL's constructions definitions. I would argue every number in a computer is inexact by definition since our Turning machines can't compute Real numbers. Still, is this just about precision, or does this choice also affect their manifoldness guarantees? And how much performance difference does it make?

Reply:
Reals are out of reach, but rationals aren’t (if you avoid sqrt): store arbitrary precision integers  for numerator and denominator, simplify fractions by finding their gcd after each operation, and you’re only limited by memory. That’s what Gmpq is doing (wrapper around [GNU MP](https://en.m.wikipedia.org/wiki/GNU_Multiple_Precision_Arithmetic_Library)), powering the Epeck kernel when ‘-DCGAL_USE_GMPXX=1’ is set.

Reply:
Re/ manifoldness guarantees Epick does magic by offering exact predicates even though numbers are ultimately represented as doubles. If the corefinement inputs are manifold (and closed), so should be the outputs.